modules/data-streams/src/main/java/org/elasticsearch/datastreams/lifecycle/DataStreamLifecycleService.java

```diff
@@ -512,7 +790,42 @@ public void onFailure(Exception e) {
         });
     }
 
-    private void deleteIndex(DeleteIndexRequest deleteIndexRequest, TimeValue retention, ActionListener<Void> listener) {
+    private void addIndexBlock(AddIndexBlockRequest addIndexBlockRequest, ActionListener<Void> listener) {
+        assert addIndexBlockRequest.indices() != null && addIndexBlockRequest.indices().length == 1
+            : "Data stream lifecycle service updates the index block for one index at a time";
+        // "saving" the index name here so we don't capture the entire request
+        String targetIndex = addIndexBlockRequest.indices()[0];
+        logger.trace(
+            "Data stream lifecycle service issues request to add block [{}] for index [{}]",
+            addIndexBlockRequest.getBlock(),
+            targetIndex
+        );
+        client.admin().indices().addBlock(addIndexBlockRequest, new ActionListener<>() {
+            @Override
+            public void onResponse(AddIndexBlockResponse addIndexBlockResponse) {
+                logger.info(
+                    "Data stream lifecycle service successfully added block [{}] for index index [{}]",
+                    addIndexBlockRequest.getBlock(),
+                    targetIndex
+                );
+                listener.onResponse(null);
+            }
```

Comment:
Out of curiosity, what should we do if the `addIndexBlockResponse` was not acknowledged, or if it has shard failures? Here we ignore any of the internals for it.

Reply:
Added code to register the failure(s). On the next DSL run we'll retry if needed